templates/frontEndLayout.html

```diff
@@ -289,16 +299,218 @@
+        state["plot" + x + y].seedType = seed;
+        state["plot" + x + y].timeLeft = GAME_CONFIG.seeds[seed].harvestTimeSeconds;
+        updateDisplay();
+        var progressBar = document.getElementById("progressBar" + x + y);
+        progressBar.style.display = "block";
+    } else {
+        window.alert("No " + GAME_CONFIG.seeds[seed].name + " seeds to plant.");
+    }
+};
+
+harvest = function(x, y) {
+    var seed = state["plot" + x + y].seedType;
+    state[seed] += GAME_CONFIG.seeds[seed].harvestYield;
+    document.getElementById("owned" + seed).innerHTML = state[seed];
+    var bigBox = document.getElementById("plot_" + x + "_" + y);
+    var medBoxes = bigBox.getElementsByClassName("container2");
+    for(i = 0; i < medBoxes.length; i++) {
+        medBoxes[i].style.display = "block";
+    }
+    var box = document.getElementById("growing" + x + y);
+    box.style.display = "none";
+    box.style.backgroundImage = "url(" + GAME_CONFIG.seeds[seed].imageLarge + ")";
+    var soil = document.getElementById("soil" + x + y);
+    soil.style.display = "block";
+    var button = document.getElementById("button" + x + y);
+    button.style.display = "none";
+    seed = 0;
+}
+
+tick = function() {
+    for(i = 0; i <= 2; i++) {
+        for(j = 0; j <= 2; j++) {
+            if(state["plot" + i + j].seedType != 0) {
+                var x = state["plot" + i + j].timeLeft;
+                var y = state["plot" + i + j].seedType;
+                var percent = x / GAME_CONFIG.seeds[y].harvestTimeSeconds
+                document.getElementById("filler" + i + j).style.width = percent * 100 + "%";
+                var clock = Math.floor(x / 60) + ":" + x % 60;
+                document.getElementById("time" + i + j).innerHTML = clock;
+                state["plot" + i + j].timeLeft --;
+                if (x == 0) {
+                    document.getElementById("progressBar" + i + j).style.display = "none";
+                    document.getElementById("button" + i + j).style.display = "block";
+                }
+
+            };
+        };
+    };
+};
+
+updateDisplay = function() {
+    for(i = 0; i <= 2; i++) {
+        for(j = 0; j <= 2; j++) {
+            if(state["plot" + i + j].seedType != 0) {
+                var x = state["plot" + i + j].timeLeft;
+                var y = state["plot" + i + j].seedType;
+                var percent = x / GAME_CONFIG.seeds[y].harvestTimeSeconds
+                document.getElementById("filler" + i + j).style.width = percent * 100 + "%";
+                var clock = Math.floor(x / 60) + ":" + x % 60;
+                document.getElementById("time" + i + j).innerHTML = clock;
+                if (x == 0) {
+                    document.getElementById("progressBar" + i + j).style.display = "none";
+                    document.getElementById("button" + i + j).style.display = "block";
+                }
+
+            };
+        };
+    };
+};
+
+
+setInterval(tick, 1000);
```

Comment:
Now for a detail I purposely left out.  setInterval doesn't actually guarantee that it'll call every 1000ms.  It does a pretty good job, but in some situations it can be much more or a bit less.

To be precise, you're going to want to check the current time against the time the seed was planted.  It's a bit more math, but it means that your game will update independent of how fast setInterval calls your function.

(this is exactly related to "slowdown" you'll experience in some games, particularly older ones-- working independent of how often your game loop (`tick`) is called means your game will just have a lower framerate but still run at the same wall-clock speed rather than slowing according to how fast the computer can run)